baseapp/grpcserver.go

```diff
@@ -74,7 +76,10 @@ func (app *BaseApp) RegisterGRPCServer(server gogogrpc.Server) {
 			newMethods[i] = grpc.MethodDesc{
 				MethodName: method.MethodName,
 				Handler: func(srv interface{}, ctx context.Context, dec func(interface{}) error, _ grpc.UnaryServerInterceptor) (interface{}, error) {
-					return methodHandler(srv, ctx, dec, interceptor)
+					return methodHandler(srv, ctx, dec, grpcmiddleware.ChainUnaryServer(
+						grpcrecovery.UnaryServerInterceptor(),
```

Comment:
I wanted to add this interceptor upstream, here: https://github.com/cosmos/cosmos-sdk/blob/b6bd4f5c2e6c00d4fe33f07a54ef482b39d3b72f/server/grpc/server.go#L16

but that doesn't catch panics unfortunately (not sure why).

Reply:
How is this displayed? Is it "user friendly"?

Reply:
There are two paths:
- direct grpc server: it shows a nice error message

```
rpc error: code = Internal desc = runtime error: invalid memory address or nil pointer dereference
```

- ABCI Query: it shows 

```
panic
```

:(. In the code, I use `sdkerrors.Wrapf(sdkerrors.ErrPanic, "%v", r)`, which shows a nice error. However, this is passed into ABCI, and it seems that Tendermint strips the message, and only shows `panic`.

Reply:
I added 80fbbfdd016c3d7a5af373ea19b9fa8050420a7e  and the error message is now:

```
panic message redacted to hide potentially sensitive system info: panic
```

lmk if that's okay @robert-zaremba @alexanderbez